src/effects/builtin/whitenoiseeffect.cpp

```diff
@@ -0,0 +1,69 @@
+#include "effects/builtin/whitenoiseeffect.h"
+
+#include <QRandomGenerator>
+
+#include "util/rampingvalue.h"
+
+// static
+QString WhiteNoiseEffect::getId() {
+    return QStringLiteral("org.mixxx.effects.whitenoise");
+}
+
+// static
+EffectManifestPointer WhiteNoiseEffect::getManifest() {
+    EffectManifestPointer pManifest(new EffectManifest());
+    pManifest->setId(getId());
+    pManifest->setName(QObject::tr("White Noise"));
+    pManifest->setAuthor("The Mixxx Team");
+    pManifest->setVersion("1.0");
+    pManifest->setDescription(QObject::tr("Creates active white noise sound"));
+
+    // This is loudness parameter
+    EffectManifestParameterPointer intensity = pManifest->addParameter();
+    intensity->setId("Loudness");
+    intensity->setName(QObject::tr("Loudness"));
+    intensity->setDescription(QObject::tr("Set the loudness which makes the noise louder or more quiet"));
+    intensity->setControlHint(EffectManifestParameter::ControlHint::KNOB_LOGARITHMIC);
+    intensity->setSemanticHint(EffectManifestParameter::SemanticHint::UNKNOWN);
+    intensity->setUnitsHint(EffectManifestParameter::UnitsHint::UNKNOWN);
+    intensity->setDefaultLinkType(EffectManifestParameter::LinkType::LINKED);
+    intensity->setMinimum(0);
+    intensity->setDefault(0);
+    intensity->setMaximum(1);
+
+    return pManifest;
+}
+
+WhiteNoiseEffect::WhiteNoiseEffect(EngineEffect* pEffect)
+        : m_pLoudnessParameter(pEffect->getParameterById("Loudness")) {
+}
+
+WhiteNoiseEffect::~WhiteNoiseEffect() {
+}
+
+void WhiteNoiseEffect::processChannel(
+        const ChannelHandle& handle,
+        WhiteNoiseGroupState* pGroupState,
+        const CSAMPLE* pInput,
+        CSAMPLE* pOutput,
+        const mixxx::EngineParameters& bufferParameters,
+        const EffectEnableState enableState,
+        const GroupFeatureState& groupFeatures) {
+    Q_UNUSED(handle);
+    Q_UNUSED(groupFeatures);
+    Q_UNUSED(enableState);
+
+    WhiteNoiseGroupState& gs = *pGroupState;
+    CSAMPLE loudness = m_pLoudnessParameter->value();
+    RampingValue<CSAMPLE_GAIN> loudness_ramping_value(
+            loudness, gs.prev_loudness, bufferParameters.framesPerBuffer());
+
+    for (unsigned int i = 0; i < bufferParameters.samplesPerBuffer(); i ++) {
+        CSAMPLE_GAIN loudness_ramped = loudness_ramping_value.getNext();
+        float output = static_cast<float>(
+                QRandomGenerator::global()->generateDouble() * loudness_ramped);
+        pOutput[i] = pInput[i] * (1 - loudness_ramped) + output;
```

Comment:
```suggestion
        float noise = static_cast<float>(
                QRandomGenerator::global()->generateDouble() * loudness_ramped);
        pOutput[i] = pInput[i] * (1 - loudness_ramped) + noise * loudness_ramped;
```
`output` could be confused with `pOutput`